Bottom: da890d35bb968ba3d6c80202522ea73e53e9572e
Top:    2f831db0d7c4d5daf8eba4acb1426ea5d6105419
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-08-25 01:40:33 +0100

HACK: Play with where the GDK version needs to draw crosshair attachments

THESE CHANGES ARE NOT WANTED, JUST A QUICK HACK TO EXPERIMENT WITH


---

diff --git a/src/hid/gtk/gtkhid-gdk.c b/src/hid/gtk/gtkhid-gdk.c
index 5b52b9a..56b7aee 100644
--- a/src/hid/gtk/gtkhid-gdk.c
+++ b/src/hid/gtk/gtkhid-gdk.c
@@ -984,8 +984,10 @@ ghid_invalidate_all ()
 
   hid_expose_callback (&ghid_hid, &region, 0);
   ghid_draw_grid ();
-  if (ghidgui->need_restore_crosshair)
+  if (ghidgui->need_restore_crosshair) {
     RestoreCrosshair (FALSE);
+    DrawAttached (FALSE);
+  }
   ghidgui->need_restore_crosshair = FALSE;
   ghid_screen_update ();
 }
diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index 7cd464a..16e1be8 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -248,6 +248,7 @@ ghid_note_event_location (GdkEventButton * ev)
   gport->view_x = event_x * gport->zoom + gport->view_x0;
   gport->view_y = event_y * gport->zoom + gport->view_y0;
 
+  DrawAttached (false);
   moved = MoveCrosshairAbsolute (SIDE_X (gport->view_x), 
 				 SIDE_Y (gport->view_y));
   if (moved)
@@ -255,6 +256,7 @@ ghid_note_event_location (GdkEventButton * ev)
       AdjustAttachedObjects ();
       RestoreCrosshair (false);
     }
+  DrawAttached (false);
   ghid_set_cursor_position_labels ();
   return moved;
 }
